test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -241,3 +258,34 @@ func extensionConfig(specName string, namespace *corev1.Namespace) *runtimev1.Ex
 		},
 	}
 }
+
+// responsesConfigMap generates a ConfigMap with preloaded responses for the test extension.
+func responsesConfigMap(name string, namespace *corev1.Namespace) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-hookresponses", name),
+			Namespace: namespace.Name,
+		},
+		// Every response contain only Status:Success. The test checks whether each handler has been called at least once.
+		Data: map[string]string{
+			"BeforeClusterCreate-response":          "{\"Status\": \"Success\"}",
+			"BeforeClusterUpgrade-response":         "{\"Status\": \"Success\"}",
+			"AfterControlPlaneInitialized-response": "{\"Status\": \"Success\"}",
+			"AfterControlPlaneUpgrade-response":     "{\"Status\": \"Success\"}",
+			"AfterClusterUpgrade-response":          "{\"Status\": \"Success\"}",
+		},
+	}
+}
+
+func checkLifecycleHooks(ctx context.Context, c client.Client, namespace string, clusterName string, hooks map[string]string) error {
+	configMap := &corev1.ConfigMap{}
+	configMapName := clusterName + "-hookresponses"
```

Comment:
What we're checking here is that each hook has been called at least once.

We're not checking to see if the hooks are blocking or not, but plan to do that in a future iteration.